src/libstd/ffi/c_str.rs

```diff
@@ -225,8 +225,11 @@ impl CString {
     /// Retakes ownership of a `CString` that was transferred to C.
     ///
     /// This should only ever be called with a pointer that was earlier
-    /// obtained by calling `into_raw` on a `CString`. Additionally, the length
-    /// of the string will be recalculated from the pointer.
+    /// obtained by calling `into_raw` on a `CString`. In particular, using this method
+    /// to create a `CString` pointing at memory that will be freed by other code
+    /// (such as a C library) will lead to undefined behavior!
```

Comment:
Well, sure, it doesn’t always lead to UB regardless of its arguments (i.e. if you immediately forget it, however unreasonable such code would be). Moreover using similar format as `String::from_raw` and `Vec::from_raw` helps consistency :)